aksharamukha/transliterate.py

```diff
@@ -517,6 +518,43 @@ def convert_default(src, tgt, txt, nativize = True, post_options = [], pre_optio
 
     return convert(src, tgt, txt, nativize, pre_options, post_options)
 
+def convert_docx(src, tgt, file, nativize, pre_options, post_options):
+    new_file = io.BytesIO()      
+
+    with ZipFile(file, 'r') as old_archive:
+        with ZipFile(new_file, 'w') as new_archive:
+            for item in old_archive.filelist:
+                if item.filename.startswith('word/') and item.filename.endswith('.xml'):
+                    with old_archive.open(item, 'r') as xml_file:
+                        tree = etree.parse(xml_file)
+                        for element in tree.getroot().iter():
+                            if element.tag == "{http://schemas.openxmlformats.org/wordprocessingml/2006/main}t":
+                                element.text = convert(
+                                    src, tgt, element.text, nativize, post_options, pre_options)
+
+                        string = etree.tostring(
+                        tree.getroot(), xml_declaration=True, encoding="UTF-8", standalone=True)
+
+                        new_archive.writestr(item, string)
+
+                else:
+                    # Copy other contents as it is
+                    new_archive.writestr(item, old_archive.read(item.filename))
+
+    if(isinstance(file, str)):
+        # python package use
+        # Flush new zip to the file
+        with open(file, 'wb') as f:
+            f.write(new_file.getvalue())
+    else:
+        # web-api call
+        # Flush new zip into new in-memory stream
+        file = io.BytesIO(new_file.getvalue())
+    
+    new_file.close()
```

Comment:
The problem is that when we close the stream, we can't return its data because it is already closed and it will throw `ValueError: I/O operation on closed file.`.I tried to use a **Thread** to handle the closing process but it didn't work. So, we have two options:
1. Leave it open and let the garbage collector handle it after the API finishes its job.
2.  return its binary data as a string, close the in-memory stream, send the string to the front-end side, and JS will write the data in a newly created file.

Reply:
I meant to use a `with` close for `new_file` so that we are sure that it will be closed even in case of run-time exceptions. As far as I can see this `new_file` object is not returned by the method `convert_docx`.